statsmodels/multivariate/factor.py

```diff
@@ -0,0 +1,245 @@
+# -*- coding: utf-8 -*-
+
+from statsmodels.base.model import Model
+from .factor_rotation import rotate_factors, promax
+from .plots import plot_scree, plot_loadings
+
+import numpy as np
+from numpy.linalg import eig, inv, norm, matrix_rank
+import pandas as pd
+
+from statsmodels.iolib import summary2
+
+
+class Factor(Model):
+    """
+    Principle axis factor analysis
+
+    .. [1] http://www.real-statistics.com/multivariate-statistics/
+    factor-analysis/principal-axis-method/
+
+    .. [2] http://stats.stackexchange.com/questions/102882/
+    steps-done-in-factor-analysis-compared-to-steps-done-in-pca/102999#102999
+
+    Supported rotations:
+        'varimax', 'quartimax', 'biquartimax', 'equamax', 'oblimin',
+        'parsimax', 'parsimony', 'biquartimin', 'promax'
+
+    Parameters
+    ----------
+    endog : array-like
+        Variables in columns, observations in rows
+
+    n_factor : int
+        The number of factors to extract
+
+    """
+    def __init__(self, endog, n_factor, exog=None, **kwargs):
+        if n_factor <= 0:
+            raise ValueError('n_factor must be larger than 0! %d < 0' %
+                             (n_factor))
+        if n_factor > endog.shape[1]:
+            raise ValueError('n_factor must be smaller or equal to the number'
+                             ' of columns of endog! %d > %d' %
+                             (n_factor, endog.shape[1]))
+        self.n_factor = n_factor
+        self.rotation = None
+        self.loadings = None
+        self.loadings_no_rot = None
+        self.communality = None
+        self.eigenvals = None
+        super(Factor, self).__init__(endog, exog)
+
+    def fit(self, n_max_iter=50, tolerance=1e-6, rotation=None,
+            SMC=True):
+        """
+        Fit the factor model
+
+        Parameters
+        ----------
+        n_max_iter : int
+            Maximum number of iterations for communality estimation
+        tolerance : float
+            If `norm(communality - last_communality)  < tolerance`,
+            estimation stops
+        rotation : string
+            rotation to be applied
+        SMC : True or False
+            Whether or not to apply squared multiple correlations
+
+        -------
+
+        """
+        if rotation not in [None, 'varimax', 'quartimax', 'biquartimax',
+                            'equamax', 'oblimin', 'parsimax', 'parsimony',
+                            'biquartimin', 'promax']:
+            raise ValueError('Unknown rotation method %s' % (rotation))
+        R = pd.DataFrame(self.endog).corr().values
```

Comment:
use numpy.corrcoef(self.endog, rowvar=0)

Reply:
as part of data, this could be an attribute.
It doesn't change in repeated calls to fit